sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersIntegrationTestBase.java

```diff
@@ -43,69 +47,67 @@
         Pattern.compile(String.format("(?:%s)(.*?)(?:\",|\"})", JSON_PROPERTIES_TO_REDACT.toString()), Pattern.CASE_INSENSITIVE);
 
     protected PhoneNumbersClientBuilder getClientBuilder(HttpClient httpClient) {
-        if (getTestMode() == TestMode.PLAYBACK) {
-            httpClient = interceptorManager.getPlaybackClient();
-        }
-
         CommunicationConnectionString communicationConnectionString = new CommunicationConnectionString(CONNECTION_STRING);
         String communicationEndpoint = communicationConnectionString.getEndpoint();
         String communicationAccessKey = communicationConnectionString.getAccessKey();
 
         PhoneNumbersClientBuilder builder = new PhoneNumbersClientBuilder();
         builder
-                .httpClient(httpClient)
+                .httpClient(getHttpClient(httpClient))
+                .addPolicy(getOverrideMSUseragentPolicy())
                 .endpoint(communicationEndpoint)
                 .credential(new AzureKeyCredential(communicationAccessKey));
 
-        if (getTestMode() == TestMode.RECORD) {
-            List<Function<String, String>> redactors = new ArrayList<>();
-            redactors.add(data -> redact(data, JSON_PROPERTY_VALUE_REDACTION_PATTERN.matcher(data), "REDACTED"));
-            builder.addPolicy(interceptorManager.getRecordPolicy(redactors));
+        if (shouldRecord()) {
+            builder.addPolicy(getRecordPolicy());
         }
 
         return builder;
     }
 
     protected PhoneNumbersClientBuilder getClientBuilderWithConnectionString(HttpClient httpClient) {
-        if (getTestMode() == TestMode.PLAYBACK) {
-            httpClient = interceptorManager.getPlaybackClient();
-        }
-
         PhoneNumbersClientBuilder builder = new PhoneNumbersClientBuilder();
         builder
-                .httpClient(httpClient)
+                .httpClient(getHttpClient(httpClient))
+                .addPolicy(getOverrideMSUseragentPolicy())
                 .connectionString(CONNECTION_STRING);
 
-        if (getTestMode() == TestMode.RECORD) {
-            List<Function<String, String>> redactors = new ArrayList<>();
-            redactors.add(data -> redact(data, JSON_PROPERTY_VALUE_REDACTION_PATTERN.matcher(data), "REDACTED"));
-            builder.addPolicy(interceptorManager.getRecordPolicy(redactors));
+
+        if (shouldRecord()) {
+            builder.addPolicy(getRecordPolicy());
         }
 
         return builder;
     }
 
     protected PhoneNumbersClientBuilder getClientBuilderUsingManagedIdentity(HttpClient httpClient) {
+        
         PhoneNumbersClientBuilder builder = new PhoneNumbersClientBuilder();
         builder
-                .endpoint(new CommunicationConnectionString(CONNECTION_STRING).getEndpoint())
-                .httpClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient);
```

Comment:
I think the logic has been changed here when you replaced the shortened if by the `getHttpClient` function.

Reply:
I found [this](https://github.com/Azure/azure-sdk-for-java/blob/3f31d68eed6fbe11516ca3afe3955c8840a6e974/sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java#L151) method doc which implies that testing for null is the same as testing for playback mode

Reply:
Is there any other reason why this function test this differently ?

Reply:
It is guaranteed that when `httpClient` *is null*, TestMode is `PLAYBACK`